rnd/autogpt_server/autogpt_server/server/integrations.py

```diff
@@ -80,6 +84,85 @@ async def callback(
     )
 
 
+@integrations_api_router.get(
+    "/{provider}/credentials", response_model=list[CredentialsMetaResponse]
+)
+async def list_credentials(
+    provider: Annotated[str, Path(title="The provider to list credentials for")],
+    user_id: Annotated[str, Depends(get_user_id)],
+    store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
+) -> list[CredentialsMetaResponse]:
+    credentials = store.get_creds_by_provider(user_id, provider)
+    return [
+        CredentialsMetaResponse(credentials_id=cred.id, credentials_type=cred.type)
+        for cred in credentials
+    ]
+
+
+class CredentialsDetailResponse(BaseModel):
+    id: str
+    provider: str
+    title: str
+    type: Literal["oauth2", "api_key"]
+    access_token: str
+    refresh_token: Optional[str]
+    access_token_expires_at: Optional[int]
+    refresh_token_expires_at: Optional[int]
+    scopes: list[str]
+    metadata: dict
+
+
+@integrations_api_router.get(
+    "/{provider}/credentials/{cred_id}", response_model=CredentialsDetailResponse
+)
+async def get_credential(
+    provider: Annotated[str, Path(title="The provider to retrieve credentials for")],
+    cred_id: Annotated[str, Path(title="The ID of the credentials to retrieve")],
+    user_id: Annotated[str, Depends(get_user_id)],
+    store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
+) -> CredentialsDetailResponse:
+    credential = store.get_creds_by_id(user_id, cred_id)
+    if not credential:
+        raise HTTPException(status_code=404, detail="Credentials not found")
+    if credential.provider != provider:
+        raise HTTPException(
+            status_code=400, detail="Credentials do not match the specified provider"
+        )
+
+    if isinstance(credential, OAuth2Credentials):
+        return CredentialsDetailResponse(
```

Comment:
Should I split this into two models, one for each type?

Reply:
Why not just use the `Credentials` union type?

Reply:
will do, I see secrets are dumped properly